<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModuleSetup</class>
 <widget class="QDialog" name="ModuleSetup">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>619</width>
    <height>387</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Module setup</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QLabel" name="textLabel1_4_2">
       <property name="text">
        <string>MCPD ID</string>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="MCPDSpinBox" name="devid">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="maximum">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="textLabel1_4_2_2">
       <property name="text">
        <string>Module
(Bus-#)</string>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="ModuleSpinBox" name="module">
       <property name="maximum">
        <number>7</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="modeGroup">
     <property name="title">
      <string>Mode</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_18">
      <item row="0" column="0">
       <widget class="QRadioButton" name="pos">
        <property name="text">
         <string>Position</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="amp">
        <property name="text">
         <string>Amplitude</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="groupBox13">
     <property name="title">
      <string>Amplifiers, Discriminator</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_10">
      <item row="0" column="0">
       <widget class="QLabel" name="textLabel1_4_2_3_2">
        <property name="text">
         <string>Gain</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="comgain">
        <property name="text">
         <string>common</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="channelLabel">
          <property name="text">
           <string>Channel</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="channel">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="maximum">
           <number>7</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="3">
       <widget class="QLabel" name="textLabel1_4_2_3_3_3">
        <property name="text">
         <string>Gain (0...255)</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="4">
       <widget class="QLineEdit" name="gain">
        <property name="text">
         <string>1</string>
        </property>
        <property name="cursorPosition">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="0" column="5">
       <widget class="QPushButton" name="setGain">
        <property name="text">
         <string>set gain</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QLabel" name="textLabel1_4_2_3_3">
        <property name="text">
         <string>Threshold</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLabel" name="textLabel1_4_2_3_3_2">
        <property name="text">
         <string>Threshold</string>
        </property>
        <property name="wordWrap">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLineEdit" name="threshold">
        <property name="text">
         <string>20</string>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QPushButton" name="setThreshold">
        <property name="text">
         <string>set &amp;threshold</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox35">
     <property name="title">
      <string>Pulser</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_19">
      <item row="0" column="0">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="pulserButton">
            <property name="toolTip">
             <string>toggle pulser on/off</string>
            </property>
            <property name="text">
             <string>Pulser</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="textLabel1_4_2_3_4">
            <property name="text">
             <string>Channel</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="pulsChan">
            <property name="toolTip">
             <string>pulser channel</string>
            </property>
            <property name="maximum">
             <number>7</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="buttonGroup5">
        <property name="title">
         <string>Amplitude (0...255)</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_17">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QRadioButton" name="pulsampRadio1">
             <property name="text">
              <string/>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="pulsAmp1">
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>50</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="pulsampRadio2">
             <property name="text">
              <string/>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="pulsAmp2">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
             <property name="value">
              <number>90</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QGroupBox" name="buttonGroup4">
        <property name="title">
         <string>Position</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_11">
         <item row="0" column="0">
          <widget class="QRadioButton" name="pulsLeft">
           <property name="text">
            <string>left</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QRadioButton" name="pulsMid">
           <property name="text">
            <string>mid</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QRadioButton" name="pulsRight">
           <property name="text">
            <string>&amp;right</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox39">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>Register op</string>
     </property>
     <widget class="QPushButton" name="readRegisterButton">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>20</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Read</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="registerSelect">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>20</y>
        <width>40</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>2</number>
      </property>
     </widget>
     <widget class="QLineEdit" name="registerValue">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>20</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="writeRegisterButton">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>81</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Write</string>
      </property>
     </widget>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ModuleSpinBox</class>
   <extends>QSpinBox</extends>
   <header>ModuleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>MCPDSpinBox</class>
   <extends>QSpinBox</extends>
   <header>MCPDSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>309</x>
     <y>193</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>channel</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>displaySlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>668</x>
     <y>679</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comgain</sender>
   <signal>toggled(bool)</signal>
   <receiver>channel</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>571</x>
     <y>679</y>
    </hint>
    <hint type="destinationlabel">
     <x>668</x>
     <y>679</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comgain</sender>
   <signal>toggled(bool)</signal>
   <receiver>channelLabel</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>571</x>
     <y>679</y>
    </hint>
    <hint type="destinationlabel">
     <x>617</x>
     <y>681</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>amp</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setModeSlot(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>447</x>
     <y>741</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>module</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>displaySlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>353</x>
     <y>739</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsampRadio1</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>781</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsampRadio1</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsAmp1</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>781</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>878</x>
     <y>920</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsampRadio1</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsAmp2</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>781</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>1073</x>
     <y>920</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsampRadio2</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>976</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsampRadio2</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsAmp2</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>976</x>
     <y>917</y>
    </hint>
    <hint type="destinationlabel">
     <x>1073</x>
     <y>920</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsChan</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>913</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulserButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>366</x>
     <y>916</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsLeft</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1209</x>
     <y>921</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsMid</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
    <x>1340</x>
     <y>921</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsRight</sender>
   <signal>toggled(bool)</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setPulserSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1470</x>
     <y>921</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>readRegisterButton</sender>
   <signal>clicked()</signal>
   <receiver>ModuleSetup</receiver>
   <slot>readRegisterSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>442</x>
     <y>796</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>writeRegisterButton</sender>
   <signal>clicked()</signal>
   <receiver>ModuleSetup</receiver>
   <slot>writeRegisterSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>352</x>
     <y>796</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setGain</sender>
   <signal>clicked()</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setGainSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1502</x>
     <y>682</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gain</sender>
   <signal>returnPressed()</signal>
   <receiver>setGain</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>844</x>
     <y>680</y>
    </hint>
    <hint type="destinationlabel">
     <x>1502</x>
     <y>682</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setThreshold</sender>
   <signal>clicked()</signal>
   <receiver>ModuleSetup</receiver>
   <slot>setThresholdSlot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1502</x>
     <y>747</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>threshold</sender>
   <signal>returnPressed()</signal>
   <receiver>setThreshold</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>844</x>
     <y>745</y>
    </hint>
    <hint type="destinationlabel">
     <x>1502</x>
     <y>747</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>displaySlot(int)</slot>
  <slot>setModeSlot(bool)</slot>
  <slot>displaySlot(int)</slot>
  <slot>setPulserSlot()</slot>
  <slot>readRegisterSlot()</slot>
  <slot>writeRegisterSlot()</slot>
  <slot>setGainSlot()</slot>
  <slot>setThresholdSlot()</slot>
 </slots>
</ui>
